lib/cli.ex

```diff
@@ -4,9 +4,22 @@ defmodule Onigumo.CLI do
   }
 
   def main(argv) do
-    {[], [component]} = OptionParser.parse!(argv, strict: [])
-    {:ok, module} = Map.fetch(@components, String.to_atom(component))
-    root_path = File.cwd!()
-    module.main(root_path)
+    try do
+      {[], [component]} = OptionParser.parse!(argv, strict: [])
+      component
+    rescue
+      _ in [OptionParser.ParseError, MatchError] -> usage_message()
+    else
+      component ->
+        {:ok, module} = Map.fetch(@components, String.to_atom(component))
+        root_path = File.cwd!()
+        module.main(root_path)
+    end
+  end
+
+  defp usage_message() do
+    IO.puts("Usage: ./onigumo [COMPONENT]\n")
+    IO.puts("\tSimple program that retrieve http web content in structured data.\n")
+    IO.puts("COMPONENT\tonigumo component to run, availables #{inspect(Map.keys(@components))}")
```

Comment:
Do we want to keep _inspect_? It exposes Elixir data structure that is only to be used for debugging. Maybe rather use `Enum.join("\n", Map.keys(@components))`?

Moreover, the expression is already pretty complex. It may deserve its own variable.